thinc/backends/_custom_kernels.cu

```diff
@@ -3,8 +3,27 @@
 // This pattern ensures that all of the loop values are visited once, no matter
 // what grid parameters are used for the function.
 
-extern "C" __global__
-void seq2col(float* output, const float* X, const int* lengths,
+// We cannot include CUDA header for mathetmatical constants, since it requires
+// that the development headers of the CUDA toolkit are installed.
+
+template <typename T>
+struct Constants {};
+
+template <>
+struct Constants<double> {
+    static constexpr double INV_SQRT_2 = 0.7071067811865475;
+    static constexpr double INV_SQRT_2PI = 0.3989422804014327;
+};
+
+template <>
+struct Constants<float> {
+    static constexpr float INV_SQRT_2 = 0.70710677;
```

Comment:
Slightly surprised that this isn't `0.70710678` considering the `double` value of this one?

Reply:
`0.70710678` is not representable using a 32-bit float.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0f9a148c5d96bf7fe081a15b10676c5b